middleware/source_ips_test.go

```diff
@@ -192,7 +192,198 @@ func TestGetSourceIPs(t *testing.T) {
+			req: &http.Request{
+				RemoteAddr: "192.168.1.100:3454",
+				Header: map[string][]string{
+					http.CanonicalHeaderKey(xForwardedFor): {"[2001:db8:cafe::17]:4711, 10.10.13.20"},
+				},
+			},
+			want: "2001:db8:cafe::17, 10.10.13.20, 192.168.1.100",
+		},
+		{
+			name: "no header, no remote",
+			req:  &http.Request{},
+			want: "",
+		},
+		{
+			name: "X-Real-IP with IPv6 remote with port",
+			req: &http.Request{
+				RemoteAddr: "192.168.1.100:3454",
+				Header: map[string][]string{
+					http.CanonicalHeaderKey(xRealIP): {"[2001:db8:cafe::17]:4711"},
+				},
+			},
+			want: "2001:db8:cafe::17, 192.168.1.100",
+		},
+		{
+			name: "X-Real-IP with IPv4 remote",
+			req: &http.Request{
+				RemoteAddr: "192.168.1.100:3454",
+				Header: map[string][]string{
+					http.CanonicalHeaderKey(xRealIP): {"192.169.1.200"},
+				},
+			},
+			want: "192.169.1.200, 192.168.1.100",
+		},
+		{
+			name: "X-Real-IP with IPv4 remote and X-Forwarded-For",
+			req: &http.Request{
+				RemoteAddr: "192.168.1.100:3454",
+				Header: map[string][]string{
+					http.CanonicalHeaderKey(xForwardedFor): {"[2001:db8:cafe::17]:4711, 10.10.13.20"},
+					http.CanonicalHeaderKey(xRealIP):       {"192.169.1.200"},
+				},
+			},
+			want: "192.169.1.200, 192.168.1.100",
+		},
+		{
+			name: "Forwarded with IPv4 remote",
+			req: &http.Request{
+				RemoteAddr: "192.168.1.100:3454",
+				Header: map[string][]string{
+					http.CanonicalHeaderKey(forwarded): {"for=192.169.1.200"},
+				},
+			},
+			want: "192.169.1.200, 192.168.1.100",
+		},
+		{
+			name: "Forwarded with IPv4 and proto and by fields",
+			req: &http.Request{
+				RemoteAddr: "192.168.1.100:3454",
+				Header: map[string][]string{
+					http.CanonicalHeaderKey(forwarded): {"for=192.0.2.60;proto=http;by=203.0.113.43"},
+				},
+			},
+			want: "192.0.2.60, 192.168.1.100",
+		},
+		{
+			name: "Forwarded with IPv6 and IPv4 remote",
+			req: &http.Request{
+				RemoteAddr: "192.168.1.100:3454",
+				Header: map[string][]string{
+					http.CanonicalHeaderKey(forwarded): {"for=[2001:db8:cafe::17]:4711,for=192.169.1.200"},
+				},
+			},
+			want: "2001:db8:cafe::17, 192.169.1.200, 192.168.1.100",
+		},
+		{
+			name: "Forwarded with X-Real-IP and X-Forwarded-For",
+			req: &http.Request{
+				RemoteAddr: "192.168.1.100:3454",
+				Header: map[string][]string{
+					http.CanonicalHeaderKey(xForwardedFor): {"[2001:db8:cafe::17]:4711, 10.10.13.20"},
+					http.CanonicalHeaderKey(xRealIP):       {"192.169.1.200"},
+					http.CanonicalHeaderKey(forwarded):     {"for=[2001:db8:cafe::17]:4711,for=192.169.1.200"},
+				},
+			},
+			want: "2001:db8:cafe::17, 192.169.1.200, 192.168.1.100",
+		},
+		{
+			name: "Forwarded returns hostname",
+			req: &http.Request{
+				RemoteAddr: "192.168.1.100:3454",
+				Header: map[string][]string{
+					http.CanonicalHeaderKey(forwarded): {"for=workstation.local"},
+				},
+			},
+			want: "workstation.local, 192.168.1.100",
```

Comment:
Can you add a test for `Forwarded` that verifies the matching is case insensitive? e.g. `For=192.168.1.1`

Reply:
Good idea, implemented.